specification/inference/_types/Results.ts

```diff
@@ -63,6 +63,17 @@ export class CompletionResult {
   result: string
 }
 
+/**
+ * The rerank result object representing a single ranked document
+ * id: the original index of the document in the request
+ * score: the score of the document relative to the query
+ * text: Optional, the text of the document, if requested
+ */
+export class RankedDocument {
+  id: string
```

Comment:
I commented in https://github.com/elastic/elasticsearch/pull/106378 that we might want to rename this field.